### DISCUSSION TITLE
openzfs 2.1/systemd 249 missing /dev/zvol links

### DISCUSSION THREAD

**Discussion by rugubara** [openzfs/zfs#12351] [Q&A]
Hello,

I started to face problems with population of /dev/zvol links. The partition links are there, but the main zvol link is missing sometimes. This condition is temporary and reboot may produce the link, but links to other zvols may be missing next boot/udevadm  settle call.

```
anton@PF16W6Y2 ~ $ ls -l /dev/zvol/fast/VM/storage/W10-UEFI-Personal/
total 0
lrwxrwxrwx 1 root root 20 июл 12 13:35 Data -> ../../../../../zd112
lrwxrwxrwx 1 root root 22 июл 12 13:35 Data-part1 -> ../../../../../zd112p1
lrwxrwxrwx 1 root root 22 июл 12 13:35 Data-part2 -> ../../../../../zd112p2
lrwxrwxrwx 1 root root 20 июл 12 13:35 swap -> ../../../../../zd128
lrwxrwxrwx 1 root root 22 июл 12 13:35 swap-part1 -> ../../../../../zd128p1
lrwxrwxrwx 1 root root 22 июл 12 13:35 swap-part2 -> ../../../../../zd128p2
lrwxrwxrwx 1 root root 21 июл 12 13:35 W10-1903-part1 -> ../../../../../zd96p1
lrwxrwxrwx 1 root root 21 июл 12 13:35 W10-1903-part2 -> ../../../../../zd96p2
lrwxrwxrwx 1 root root 21 июл 12 13:35 W10-1903-part3 -> ../../../../../zd96p3
lrwxrwxrwx 1 root root 21 июл 12 13:35 W10-1903-part4 -> ../../../../../zd96p4
lrwxrwxrwx 1 root root 21 июл 12 13:35 W10-1903-part5 -> ../../../../../zd96p5
```

Note that W10-1903 has part links but no zvol link. Other 2 zvols in fast/VM/storage/W10-UEFI-Personal look OK. These zvol links are referenced in my VM config, so the VM refuses to start.

/dev/zd96 exists.

**Comment by rincebrain** [comment]
Which distro and exact version of systemd is this happening on? (I know you said 241, but unless you're building it from source yourself, usually distros stick their own versioning on...)

**Reply by rugubara** [reply]
This is Gentoo - so I'm building from the source. Gentoo follows upstream versioning. The exact version is 249-r1. 249 is no longer in the package tree - I can't compare and see what is changed.

**Reply by rugubara** [reply]
I found the reason. This was the outdated initramfs. It contained the previous version of systemd. I re-generated it and the problem went away.

**Comment by rugubara** [comment]
I found the reason. This was the outdated initramfs. It contained the previous version of systemd. I re-generated it and the problem went away.

**Comment by rugubara** [comment]
I continue to have these problems. I'd be grateful for the debugging hints.

**Reply by jgottula** [reply]
Sounds like it's probably related to #12301.
